### PR TITLE
27 add event

### PR DESCRIPTION
**Pull request by adriannebulao**

### CODE REVIEW

File: src/event/event.service.ts [hunk 1]
```diff
@@ -1,4 +1,72 @@
...
+            name,
+            description,
+            registration_link,
+            start_time: new Date(start_time),
+            end_time: new Date(end_time),
+            location_id,
+          },
+        });
+
+        if (posters && posters.length > 0) {
```
Comment (miggyalino) [inline src/event/event.service.ts:46]: use isEmpty Util function. `if (posters && !isEmpty(posters)`

File: src/event/event.service.ts [hunk 2]
```diff
@@ -1,4 +1,72 @@
...
+          await prisma.poster.createMany({
+            data: posters.map((poster) => ({
+              event_id: newEvent.id,
+              image_url: poster.image_url,
+              description: poster.description,
+            })),
+          });
+        }
```
Comment (miggyalino) [inline src/event/event.service.ts:54]: `const posterData = posters.map(({image_url, description}) => {
     event_id: newEvent.id,
     image_url,
     description
})`

then set posterData for as data for createMany

**Review by SaaammmyyyS — COMMENTED** [review]
Create a bucket in Supabase to store the poster image files. Also, add a seeder to ensure the bucket is created automatically during setup.

File: src/event/create-event.dto.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  registration_link: string;
+
+  @IsISO8601()
+  @IsNotEmpty()
+  start_time: string;
+
+  @IsISO8601()
+  @IsNotEmpty()
+  end_time: string;
+
```
Comment (SaaammmyyyS) [inline src/event/create-event.dto.ts:24]: Pa add din pala ako validation to check that `start_time` is earlier than `end_time`

File: prisma/seed.ts [hunk 4]
```diff
@@ -5,6 +5,43 @@ import { faker } from '@faker-js/faker';
...
+        throw new Error(`Failed to create image bucket: ${error.message}`);
+      }
+
+      console.log('Image bucket created successfully:', data);
+    } else {
+      console.log('Image bucket already exists');
+    }
+  } catch (error) {
+    console.error('Error managing image bucket:', error);
+  }
```
Comment (miggyalino) [inline prisma/seed.ts:42]: replace this with return messages or exceptions

File: prisma/seed.ts [hunk 5]
```diff
@@ -5,6 +5,43 @@ import { faker } from '@faker-js/faker';
 const prisma = new PrismaService();
 const supabase = new SupabaseService();
```
Comment (SaaammmyyyS) [inline prisma/seed.ts:7]: make sure na naka private yung bucket

File: src/event/event.controller.ts [hunk 6]
```diff
@@ -1,7 +1,39 @@
...
+  @UseGuards(AuthGuard)
+  @UseInterceptors(FileFieldsInterceptor([{ name: 'poster_images' }]))
+  async create(
+    @Body() createEventDto: CreateEventDto,
+    @UploadedFiles() files: { poster_images?: Express.Multer.File[] },
+  ) {
+    createEventDto.location_id = Number(createEventDto.location_id);
+
+    try {
+      const createdEvent = await this.eventService.createEvent(
```
Comment (SaaammmyyyS) [inline src/event/event.controller.ts:28]: Removed the unused variable `createdEvent`

**Review by SaaammmyyyS — COMMENTED** [review]
- [ ] Since nagpa remove tayo ng attribute sa schema the seeder should also be updated. Pa remove nlng ako ng `description` sa seedPosters thankss

File: supabase/supabase.service.ts [hunk 7]
```diff
@@ -48,6 +48,29 @@ export class SupabaseService {
 
     return getUrl.data.publicUrl;
   }
+
+  async uploadPosterToBucket(file: Express.Multer.File): Promise<string> {
+    const fileName = generateUniqueFileName(file.originalname);
+    const { data, error } = await this.supabase.storage
```
Comment (SaaammmyyyS) [inline supabase/supabase.service.ts:54]: ensure that the bucket name for poster uploads is also configurable through the .env file

File: src/event/event.service.ts [hunk 8]
```diff
@@ -1,4 +1,84 @@
...
+              data: {
+                event_id: newEvent.id,
+                image_url: posterUrl,
+              },
+            });
+          }
+        }
+
+        return newEvent;
+      });
```
Comment (miggyalino) [inline src/event/event.service.ts:69]: Can you revise this for query optimization lang. For example, we have 3 files to upload. Sa current implementation mo, file upload 2 will not start unless file upload 1 is finished, then file upload 3 will wait for file upload 2 to finish before starting the upload. To optimize this query we can batch the query para mas optimized sya. You can do something like this:

`if (files && !isEmpty(files)) {
  const uploadPromises = files.map(async (file) => {
    const uniqueFileName = `${Date.now()}-${file.originalname}`;
    const posterUrl = await this.supabaseService.uploadPosterToBucket(file);
    return { event_id: newEvent.id, image_url: posterUrl };
  });

  const posterData = await Promise.all(uploadPromises);
  await prisma.poster.createMany({ data: posterData });
}`

File: src/event/event.service.ts [hunk 9]
```diff
@@ -1,4 +1,85 @@
...
+
+      return event;
+    } catch (error) {
+      if (error instanceof HttpException) {
+        throw error;
+      }
+
+      console.error('Error creating event:', error);
```
Comment (SaaammmyyyS) [inline src/event/event.service.ts:78]: Remove test logs

File: src/event/event.controller.ts [hunk 10]
```diff
@@ -1,7 +1,36 @@
...
+import { FileFieldsInterceptor } from '@nestjs/platform-express';
 
-@Controller('event')
+@Controller('/events')
 export class EventController {
   constructor(private readonly eventService: EventService) {}
+
+  @Post()
+  @UseGuards(AuthGuard)
+  @UseInterceptors(FileFieldsInterceptor([{ name: 'poster_images' }]))
```
Comment (SaaammmyyyS) [inline src/event/event.controller.ts:20]: Should validate that the uploaded files for `[poster_images]` in valid image formats else throw an error

**Review by SaaammmyyyS — APPROVED** [review]
lgtm

**Pull request merged**